src/frontend/src/flows/recovery/confirmSeedPhrase.ts

```diff
@@ -0,0 +1,157 @@
+import { html, TemplateResult } from "lit-html";
+import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { withRef, renderPage } from "../../utils/lit-html";
+import { Chan } from "../../utils/utils";
+import { ref, createRef, Ref } from "lit-html/directives/ref.js";
+import { mainWindow } from "../../components/mainWindow";
+import { I18n } from "../../i18n";
+
+import copyJson from "./confirmSeedPhrase.json";
+
+// A list of words, where "check" indicates if the user needs to double check (re-input) a word
+type Word = { word: string } & (
+  | { check: false }
+  | { check: true; elem: Ref<HTMLInputElement> }
+);
+
+// A list of indices nicely spread over the 25 words (anchor + 24 BIP39)
+export const checkIndices = [3, 7, 12, 13, 15, 17, 18, 20, 22, 24];
+
+const checkWord = (word: Word): boolean =>
+  word.check
+    ? withRef(word.elem, (elem) => elem.value === elem.dataset.expected) ===
+      true
+    : true;
+
+const confirmSeedPhraseTemplate = ({
+  words: words_,
+  confirm,
+  i18n,
+}: {
+  words: Omit<Word, "elem">[];
+  confirm: () => void;
+  i18n: I18n;
+}) => {
+  const copy = i18n.i18n(copyJson);
+
+  // All words, where a `Ref` was added if the word needs checking
+  // NOTE: typescript can't follow if word is deconstructed with {...word}
+  const words: Word[] = words_.map((word) => {
+    if (word.check) {
+      const elem: Ref<HTMLInputElement> = createRef();
+      return { word: word.word, check: word.check, elem };
```

Comment:
nth: destructing here would be more readable because we would avoid `word.word`

Reply:
>   // NOTE: typescript can't follow if word is deconstructed with {...word}

guess the comment is a little too far then :)